_posts/2021/2021-03-08-turn-off-the-bits-of-rails-you-dont-use.md

```diff
@@ -0,0 +1,87 @@
+---
+title: "Turn off the bits of Rails you don't use"
+description: "Reduce complexity, reduce overhead"
+layout: article
+category: ruby
+image:
+  base: "2021/turn-off-the-bits-of-rails-you-dont-use"
+  alt: "Purple lit bank of audio switches"
+  source: "https://unsplash.com/photos/n4BDkIEls78"
+  credit: "Denisse Leon"
+---
+
+The Rails Framework consists of multiple subsystems that work together. The names of these systems typically begin with "Action" or "Active" e.g. Active Record, Active Support, Action Pack etc.
+
+You can see the individual component gems as directories in the [`rails/rails` repo on github](https://github.com/rails/rails).
+
+There has been a continual process of adding new, useful libraries to Rails ever since its release. For example, the addition of web socket support via Action Cable was a major feature of Rails 5. Despite the addition of large new sub-frameworks there remains a focus on modularity since [Merb was merged into Rails](https://yehudakatz.com/2008/12/23/rails-and-merb-merge/) in version 3. (This was big Ruby news at the time!)
```

Comment:
This paragraph seems a little confused. I get what you're saying...but it's kinda like: why are you saying it here (mainly the latter part of it).

Reply:
It's just context for why Rails is modular. It wasn't always. Lots of folk don't know it is.

Reply:
Yes, I think maybe it could be said more explicitly, is all. Or maybe a reason as to why the modularity was a focus. Or maybe just a slight rewording, which might involve taking out the Merb bit..or: Rails has always had a focus on modularity, stemming back from when Merb.

It might just be that the wording seemed a little of when I was reading it through last time!